src/server/wesnothd/game.cpp

```diff
@@ -522,49 +487,45 @@ void game::transfer_side_control(const socket_ptr& sock, const simple_wml::node&
 	if(side_num < 1 || side_num > sides_.size()) {
 		std::ostringstream msg;
 		msg << "The side number has to be between 1 and " << sides_.size() << ".";
-		send_server_message(msg.str(), sock);
+		send_server_message(msg.str(), player);
 		return;
 	}
 
 	if(side_num > get_sides_list().size()) {
-		send_server_message("Invalid side number.", sock);
+		send_server_message("Invalid side number.", player);
 		return;
 	}
 
 	const simple_wml::string_span& newplayer_name = cfg["player"];
-	const socket_ptr old_player = sides_[side_num - 1];
-	const auto oldplayer = player_connections_.find(old_player);
+	auto old_player = sides_[side_num - 1];
 	const std::string& controller_type = cfg["to"].to_string();
-	if(oldplayer == player_connections_.end()) {
-		missing_user(old_player, __func__);
-	}
 
-	const std::string old_player_name = username(old_player);
+	const std::string old_player_name = old_player ? username(old_player.value()) : "null";
 
 	// Not supported anymore.
 	if(newplayer_name.empty()) {
 		std::stringstream msg;
 		msg << "Received invalid [change_controller] with no player= attribute specified";
 		DBG_GAME << msg.str() << "\n";
-		send_server_message(msg.str(), sock);
+		send_server_message(msg.str(), player);
 		return;
 	}
 
 	// Check if the sender actually owns the side he gives away or is the host.
-	if(!(sock == old_player || sock == owner_)) {
+	if(!(player == old_player || player == owner_)) {
 		std::stringstream msg;
 		msg << "You can't give away side " << side_num << ". It's controlled by '" << old_player_name << "' not you.";
 		DBG_GAME << msg.str() << "\n";
-		send_server_message(msg.str(), sock);
+		send_server_message(msg.str(), player);
 		return;
 	}
 
 	// find the player that is passed control
-	socket_ptr newplayer = find_user(newplayer_name);
+	auto newplayer { find_user(newplayer_name) };
```

Comment:
Is there an issue with using `auto newplayer = find_user(newplayer_name)`?

Reply:
Getting in the habit of using brace initialization for clarity, to avoid the "most vexing parse" issue.

Reply:
I mean since we now can use brace initialization there's no need for copy-initialization.

Reply:
Is this a general change we'll employ in the complete code base? Should this be added to the coding standards/guide lines?

Reply:
Isn't this still copy initialization, just now with the copy constructor rather than the copy assignment operator?

Reply:
Aren't both copy constructors? `classname foo = bar;` does not default-construct foo followed by copy assignment, it calls the copy constructor directly. That is, it is very much not the same as `classname foo; foo = bar;`.

That means that `auto foo = bar;` and `auto foo { bar };` should call the same constructor, but there might be different parsing details. (maybe initializer lists could get involved)

Reply:
It should generally work in the same way due to copy elision. I'm still a bit torn on what of those two ways I prefer. I find that assignment annoying conceptually because it is not real assignment. Brace initialization shows the intent better. But I don't insist that entire codebase should be revamped to brace initialization. Assignment looks closer to classic C ways of doing things and using it might be a habit that is hard to get rid of.

Reply:
Personally, I tend to prefer assignment initialization just because it's more akin to how most languages do things, but I have no objection to brace-init.

Also, I must say, once one starts debating copy elision that's a sign we've descended into the C++ weeds 😂

Reply:
I would suggest to decide one way or another or describe clear rules on when to use one or the other. I don't think it's useful to just mix and match arbitrarily.